components/projects/workspace/ProjectWorkspaceView.tsx

```
@@ -615,6 +621,34 @@ export function ProjectWorkspaceView({
   const selectedDrawing =
     drawings.find((d) => d.id === selectedDrawingId) ?? null;
 
+  // Current-round totals: sum of marks that aren't yet claimed by a finished
+  // round and aren't rebar-tab reference lengths. Recomputed from the actual
+  // mark list on every render, so undo/redo can never leave it stale.
+  const sessionTotals = useMemo(() => {
+    let count = 0,
+      length = 0,
+      area = 0;
+    for (const m of measurementHook.state.measurements) {
+      if (claimedMarkIds.has(m.id) || rebarMarkIds.current.has(m.id)) continue;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/components/projects/workspace/ProjectWorkspaceView.tsx:632:39
  630 |       area = 0;
  631 |     for (const m of measurementHook.state.measurements) {
> 632 |       if (claimedMarkIds.has(m.id) || rebarMarkIds.current.has(m.id)) continue;
      |                                       ^^^^^^^^^^^^^^^^^^^^ Passing a ref to a function may read its value during render
  633 |       if (m.type === "count") count++;
  634 |       else if (m.type === "length") length += m.realLength;
  635 |       else if (m.type === "area") area += m.realArea;

(react-hooks/refs)